pkcs5/src/pbes2/encryption.rs

```diff
@@ -48,6 +49,104 @@ fn cbc_decrypt<'a, C: BlockCipherDecrypt + BlockCipher + KeyInit>(
         .map_err(|_| Error::EncryptFailed)
 }
 
+fn aes_128_gcm_encrypt<'a>(
```

Comment:
You can probably write this and `aes_*_gcm_decrypt` generically in terms of the `AesGcm` struct by parameterizing it over the `Aes` cipher used, e.g.

```rust
fn aes_gcm_encrypt<'a, Aes: BlockCipher + BlockSizeUser<BlockSize = U16> + BlockEncrypt>
```

(though I would probably recommend using a `where` clause)

If you don't want to make it generic yourself, I can probably do that as a followup.

Reply:
OK I'll give this a try

Reply:
I made an attempt at this and am hitting trait errors that don't really make sense. I think it might be caused by the existence of two different versions of `cipher` crate in the dependency tree.

Reply:
you might need to use a pre-release version of aes and aes-gcm.

```
[dependencies]
aes = { version = "=0.9.0-pre" }
aes-gcm = { version = "=0.11.0-pre" }
```

and in the root `Cargo.toml`:
```
[patch.crates-io]
aes-gcm = { git = "https://github.com/RustCrypto/AEADs.git" }
```

Reply:
Using the pre-release crates helped. Pushed c5280f31cb2597a5944a9d2573659751e3aa7765 everything seems to work